src/client/api/event.c

```diff
@@ -520,6 +520,13 @@ ev_progress_cb(void *arg)
 	if (evx->evx_nchild_running > 0)
 		return 0;
 
+	/*
+	 * If it's a thread private event but it's not owned by this thread, don't change the status
+	 * as there could be race with the thread owner.
+	 */
+	if (evx->is_th_private && !daos_event_is_priv(daos_evx2ev(evx)))
```

Comment:
how does it know it's not owned by this thread?

Reply:
I do assume that:
static __thread daos_event_t	ev_thpriv;
would be unique for each thread

Reply:
we probably can test if soak can pass if without this change?
just feel not quite straight forward why require the event cannot be set as ready by another thread, could it cause any potential problem?

Reply:
we tested without this change and it failed on 2.2
my theory is that two threads come in here;
main thread gets lock first, marks it as ready; other thread is in lock.
main thread continues to unlock and reinit (while other thread is still stuck in lock - being slow).
main thread reuses the event and makes the event in progress / running state.
other thread finally exits the lock and sees it's running and sets it as ready.

Reply:
yes i think i messed up with the checks there and mistaken it with the earlier check. you are correct. 
please ask Maureen to run 2.2 with only your PR and let's see.